report/node.go

```diff
@@ -80,10 +79,26 @@ func (n Node) WithLatest(k string, ts time.Time, v string) Node {
 	return n
 }
 
-// WithCounters returns a fresh copy of n, with Counters c merged in.
-func (n Node) WithCounters(c map[string]int) Node {
-	n.Counters = n.Counters.Merge(Counters{}.fromIntermediate(c))
-	return n
+// LookupCounter returns the value of a counter
+// (counters are stored as strings, to keep the data structure simple)
+func (n Node) LookupCounter(k string) (value int, found bool) {
+	name := CounterPrefix + k
+	var str string
+	if str, found = n.Latest.Lookup(name); found {
+		value, _ = strconv.Atoi(str)
+	}
+	return value, found
+}
+
+// WithCounter returns a fresh copy of n, with Counter c added in.
+// (counters are stored as strings, to keep the data structure simple)
+func (n Node) WithCounter(k string, value int) Node {
+	name := CounterPrefix + k
+	if prevStr, found := n.Latest.Lookup(name); found {
+		prevValue, _ := strconv.Atoi(prevStr)
+		value += prevValue
+	}
```

Comment:
Why not use `LookupCounter` here to further simplify it?

Reply:
Possibly because I thought computing `name := CounterPrefix + k` twice would slow it down, but this doesn't appear to be the case. Maybe the compiler can inline `LookupCounter`.
So done.